storage/session_memcached.go

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (C) 2024 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package storage
+
+import (
+	"strings"
+	"time"
+
+	"github.com/bradfitz/gomemcache/memcache"
+	"github.com/eko/gocache/lib/v4/cache"
+	"github.com/eko/gocache/lib/v4/store"
+	memcachestore "github.com/eko/gocache/store/memcache/v4"
+)
+
+type MemcachedSessionDatabase struct {
+	client     *memcache.Client
+	underlying *cache.Cache[[]byte]
+}
+
+// NewMemcachedSessionDatabase creates a new MemcachedSessionDatabase using an initialized memcache.Client.
+func NewMemcachedSessionDatabase(client *memcache.Client) *MemcachedSessionDatabase {
+	memcachedStore := memcachestore.NewMemcache(client, store.WithExpiration(defaultSessionDataTTL))
```

Comment:
Shouldn't we use the same value for access token lifetime? Or at least, the token lifetime shouldn't be longer than this default. Having 2 seeminly unrelated constants (but secretly relating) seems rather dangerous?

Reply:
I added a comment. Memcached requires a default setting. We override this value when writing with the TTL of the store.